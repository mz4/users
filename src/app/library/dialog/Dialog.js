import { useRef } from 'react';
import { useClickOutside } from '../../hooks';
import styles from './Dialog.module.scss';

export const Dialog = ({ header, body, footer, handleClickOutside }) => {
  const dialogRef = useRef();
  useClickOutside(dialogRef, handleClickOutside);
  return (
    <div className={styles.pageContainer}>
      <div className={styles.container} ref={dialogRef}>
        <div className={styles.header}>{header}</div>
        <div className={styles.body}>{body}</div>
        <div className={styles.footer}>{footer}</div>
      </div>
    </div>
  );
};
