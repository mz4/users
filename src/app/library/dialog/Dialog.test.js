import renderer from 'react-test-renderer';
import { Dialog } from './Dialog';

describe('Render Avatar component', () => {
  const handleClickOutside = jest.fn();
  const dialog = (
    <Dialog
      header={<div>header</div>}
      body={<div>body</div>}
      footer={<div>footer</div>}
      handleClickOutside={handleClickOutside}
    />
  );
  it('Create a snapshot', () => {
    const component = renderer.create(dialog);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
