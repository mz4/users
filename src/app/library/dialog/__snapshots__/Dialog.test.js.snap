// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Avatar component Create a snapshot 1`] = `
<div
  className="pageContainer"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <div>
        header
      </div>
    </div>
    <div
      className="body"
    >
      <div>
        body
      </div>
    </div>
    <div
      className="footer"
    >
      <div>
        footer
      </div>
    </div>
  </div>
</div>
`;
